packages/flame/lib/src/effects/provider_interfaces.dart

```diff
@@ -50,3 +50,9 @@ abstract class SizeProvider {
   Vector2 get size;
   set size(Vector2 value);
 }
+
+/// Interface for a component that can be affected by opacity effects.
```

Comment:
It would be good to document that setting the opacity to some `x` would not necessarily produce the same `x` when reading it back -- the implementation is allowed to store the opacity as an integer `alpha` which would incur some rounding errors.

Reply:
This sounds more like implementation details. No need for the interface to make any comments on that.

Reply:
Indeed, it's an implementation detail. But it is a non-obvious detail. So the interface should tell the implementers that this is something which is allowed; and to the consumers that this is something that they could expect when using this API.

Reply:
Done. Added the requested comments.